Kernel/TTY/TTY.cpp

```diff
@@ -310,13 +308,27 @@ int TTY::ioctl(FileDescription&, unsigned request, FlatPtr arg)
         if (pgid <= 0)
             return -EINVAL;
         InterruptDisabler disabler;
-        auto process = Process::from_pid(pgid.value());
+        auto process_group = ProcessGroup::from_pgid(pgid);
+        // Disallow setting a nonexistent PGID.
+        if (!process_group)
+            return -EINVAL;
+
+        auto process = Process::from_pid(ProcessID(pgid.value()));
         SessionID new_sid = process ? process->sid() : Process::get_sid_from_pgid(pgid);
         if (!new_sid || new_sid != current_process.sid())
             return -EPERM;
         if (process && pgid != process->pgid())
             return -EPERM;
-        m_process = process ? process->make_weak_ptr() : WeakPtr<Process>();
+        m_pg = process_group->make_weak_ptr();
+        if (process) {
+            auto parent = Process::from_pid(process->ppid());
+            if (parent)
```

Comment:
```suggestion
            if (auto parent = ...)
```

Reply:
Never in my life would I have thought _you_ would suggest that syntax!